classes/Attachment.php

```diff
@@ -144,6 +167,30 @@ public static function getAttachments($idLang, $idProduct, $include = true)
 				WHERE id_product = '.(int) $idProduct.'
 			)'
         );
+    }
+
+    /**
+     * Unassociate all products from the current object
+     *
+     * @param bool $update_cache
+     *            If set to true attachment cache will be updated
+     * @return bool Deletion result
+     */
+    public function deleteAttachments($update_attachment_cache = true)
+    {

```

Comment:
`$product_ids` is only used when `update_attachment_cache` is true, if not then the query is useless
suggestion:
```php
        $product_ids = array();
        if (isset($update_attachment_cache) && (bool) $update_attachment_cache === true) {
            $product_ids = Db::getInstance()->executeS('
			     SELECT `id_product` FROM `' . _DB_PREFIX_ . 'product_attachment`
			     WHERE `id_attachment` = ' . (int) $this->id);
        }

        $res = Db::getInstance()->execute('
			DELETE FROM `' . _DB_PREFIX_ . 'product_attachment`
			WHERE `id_attachment` = ' . (int) $this->id);
      
        foreach ($product_ids as $product_id) {
             Product::updateCacheAttachment((int) $product_id);
        }
```